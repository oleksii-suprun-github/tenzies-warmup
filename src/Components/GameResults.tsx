import React from 'react';
import RollDiceButton from './UI/RollDiceButton';
import Header from './UI/Header';

interface GameResults {
  timeTotal: number;
  clicksTotal: number;
  restartHandler: void;
}

const GameResults: React.Fc = ({ timeTotal, clicksTotal, restartHandler }): GameResults => {
  return (
    <>
      <Header title="Congratulation!" description="You won the game!" />
      <div id="results">
        <p>
          Total Time: <strong>{timeTotal} seconds</strong>
        </p>
        <p>
          Total Clicks: <strong>{clicksTotal} clicks</strong>
        </p>
      </div>
      <RollDiceButton onClick={restartHandler}>Start again</RollDiceButton>
    </>
  );
};
export default GameResults;
