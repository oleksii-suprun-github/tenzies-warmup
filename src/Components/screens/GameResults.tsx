import { MouseEventHandler } from 'react';
import { useTranslation } from 'react-i18next';
import RollDiceButton from '../RollDiceButton';
import Header from '../Header';

export interface GameResultsProps {
  timeTotal: number;
  clicksTotal: number;
  restartHandler: MouseEventHandler<HTMLButtonElement>;
}

const GameResults = ({ timeTotal, clicksTotal, restartHandler }: GameResultsProps) => {
  const { t } = useTranslation();
  return (
    <>
      <Header>
        <h1 className="mb-6 text-4xl font-extrabold leading-none">
          {t('game.congratulation.headline')}
        </h1>
        <p className="text-lg">{t('game.congratulation.description')}</p>
      </Header>
      <div id="results" className="mx-auto my-0 mt-[50px]">
        <p className="mb-2.5">
          {t('game.congratulation.totalTime')}:&nbsp;
          <strong>
            {timeTotal} {t('seconds')}
          </strong>
        </p>
        <p>
          {t('game.congratulation.totalClicks')}:&nbsp;
          <strong>
            {clicksTotal} {t('clicks')}
          </strong>
        </p>
      </div>
      <RollDiceButton onClick={restartHandler}>{t('game.reset')}</RollDiceButton>
    </>
  );
};

export default GameResults;
