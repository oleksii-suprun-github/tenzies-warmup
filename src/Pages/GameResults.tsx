import React from 'react';
import RollDiceButton from '../Components/RollDiceButton';
import Header from '../Components/Header';

interface GameResultsProps {
  timeTotal: number;
  clicksTotal: number;
  restartHandler: void;
}

const GameResults: React.Fc = ({ timeTotal, clicksTotal, restartHandler }): GameResultsProps => (
  <>
    <Header>
      <h1>Congratulation</h1>
      <p>You won the game!</p>
    </Header>
    <div id="results">
      <p>
        Total Time: <strong>{timeTotal} seconds</strong>
      </p>
      <p>
        Total Clicks: <strong>{clicksTotal} clicks</strong>
      </p>
    </div>
    <RollDiceButton onClick={restartHandler}>Start again</RollDiceButton>
  </>
);
export default GameResults;
