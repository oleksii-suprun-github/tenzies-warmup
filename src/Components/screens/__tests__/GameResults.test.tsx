import { render } from '@testing-library/react';
import GameResults, { GameResultsProps } from '../GameResults';
import GameResultsMockProps from './__mocks__/GameResultsProps.json';

describe('GameResults', () => {
  it('should render a component with default props', () => {
    const { container } = render(<GameResults {...(GameResultsMockProps as GameResultsProps)} />);
    expect(container).toMatchSnapshot();
  });
});
