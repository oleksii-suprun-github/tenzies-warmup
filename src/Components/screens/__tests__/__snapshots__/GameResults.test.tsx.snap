// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameResults > should render a component with default props 1`] = `
<div>
  <header
    class="text-center"
    id="header"
  >
    <h1
      class="mb-6 text-4xl font-extrabold leading-none"
    >
      Congratulations
    </h1>
    <p
      class="text-lg"
    >
      You won the game!
    </p>
  </header>
  <div
    class="mx-auto my-0 mt-[50px]"
    id="results"
  >
    <p
      class="mb-2.5"
    >
      Total Time
      : 
      <strong>
        19
         
        seconds
      </strong>
    </p>
    <p>
      Total Clicks
      : 
      <strong>
        9
         
        clicks
      </strong>
    </p>
  </div>
  <button
    class="btn btn-primary  tooltip-top mx-auto my-[25px] mt-[50px] w-fit min-w-32 bg-main-button text-2xl text-main-die"
    data-testid="roll-dice-button"
    data-tip="Please select all equal dice before rolling."
  >
    Start again
  </button>
</div>
`;
